### ISSUE TITLE
Yabridge doesn't work in Kubuntu 24.10 in Ardour, Bitwig Studio and Presonus Studio One

### ISSUE THREAD

**Issue by Skygge666** [robbert-vdh/yabridge#377]
### Thanks for giving yabridge a shot!

- [X] I read through both the troubleshooting and the known issues sections, and my issue wasn't listed there

### Problem description

In the new (K)ubuntu 24.10 (Oracular oriole) all plugins I have (MTPower Drumkit, EZDrummer 3) cause these DAWs to crash: Ardour, Bitwig Studio and Studio One. Before system upgrade, under 24.04 everything was fine. Now If I add one of these plugins or try to rescan plugins, these DAWs take whole RAM (32G in my case) then crash. What's strange, under Reaper or Tracktion Waveform the plugins are working ok. Idk what happened. I waited for new wine release and proper  Oracular repo, what happened today, but it changed nothing. Attached screenshot from Studio One but it looks the same for Ardour and Bitwig Studio.

![obraz](https://github.com/user-attachments/assets/c678a343-5d15-4ed5-9cf7-21163488e64c)

### What did you expect to happen?

I expected to load and run the plugins

### What actually happened?

Ardour, Bitwig and Studio One start to consume all RAM available then crash.

### Operating system

Kubuntu 24.10 (Oracular oriole)

### How did you install yabridge?

binary from releases page

### yabridge version

5.1.0

### yabridgectl version

5.1.0

### Wine version

winehq-staging 9.20~oracular-1

### Plugin

MT Power Drumkit, EZDrummer 3 (demo)

### Plugin type

VST3

### Plugin architecture

64-bit

### Host

Bitwig Studio, Ardour 8.10, Presonus Studio One 6.6

### Desktop environment or WM

KDE Plasma 6.1 (Wayland)

### GPU model

NVidia RTX 3060

### GPU drivers and kernel

Nvidia drivers 555, Kernel 6.11-lowlatency

**Comment by vudone** [comment]
Same problem here, in ubuntu 24.10.

I have just updated last night.. The wine-hq repositories (for oracular, the name of ubuntu24.10) installs winehq-9.20. I tried to downgrade winehq but it does not repair the problem. 

In my case when i launch Ardour (with any yabridge plugin) it keep freezie. If i launch a new ardour project, ir works fine. In any moment I load a yabridge plugin, Ardour freezie.

**Comment by vudone** [comment]
I tried to debug the code with gdb. This is whats gdb shows when the software keep Freezie. Seems like is the moment when i load Melodyne with yabridge. Im dont know this screenshots are usefull. 


![Captura desde 2024-10-21 10-12-56](https://github.com/user-attachments/assets/9aa1dde8-5adf-4013-9758-08c32116ac8d)

![Captura desde 2024-10-21 10-15-21](https://github.com/user-attachments/assets/34e1ac86-b2e3-4e6e-914b-6a219b3bf1e8)

![Captura desde 2024-10-21 10-15-21](https://github.com/user-attachments/assets/689060e8-22eb-445f-9f04-590021019537)

**Comment by vudone** [comment]
The yabridge binary release didnt work in my ubuntu 24.10. I have compiled it in my computer (with winehq-9.20) and now seems to work

**Comment by Skygge666** [comment]
Self-compiled Yabridge seems to work.

**Comment by mahlonsmith** [comment]
Also just updated to Ubuntu 24.10, with the same problem under Ardour.

Another interesting note:  Everything seemed fine using Carla as the plugin host.   Memory bloat and scanning timeouts in Ardour.

Building from source, all good in both environments.

**Comment by paulopinheiro** [comment]
I confirm the bug in Ubuntu 24.10. The problem does not occur on Carla, Reaper, Muse, Waveform13 or even Audacity. It happens with Ardour and Mixbus.
When you try to load the plugin, Yabridge isn't even called. Ardour is probably in a loop trying to load somehow, which increases memory and CPU use.
That also happens when you try to scan the plugin as a new one to add to the Ardour cache list.
But there's one more piece of information about it: if you try to scan the same plugin in the command line with ardour-vst-scanner <vst-localization> it adds to the Ardour cache with no issues. So maybe it's something about the shell environment?

**Comment by paulopinheiro** [comment]
Unfortunately, I was unable to compile yabridge from source. The process ends without errors, but at the end I don't have all the expected generated files (there is no yabridgectl, for example).

**Comment by vudone** [comment]
Hi @paulopinheiro. Im not in home now, but i think that yabridge works for u.  Yesterday when I compiled yabridge i had to compile 2 packages really. 

In the root folder, with meson and ninja, u compile some files. thats ok, its part of the project. 

In orther to compile the yabridgectl bin u need to access to the YABDRIGEFOLDER/tools/yabridgectl folder. 

and there launch: 

$ cargo run --release

yabdrigectl will be compiled. Copy it with the rest of yabridge files (probably ~/.local/share/yabridge)

And thats all.

**Comment by eugenialoli** [comment]
I hope the developer provides us with a new ubuntu binary package soon. There isn't one for the latest released version and compiling it from source is a big complicated on my system.

**Comment by Skygge666** [comment]
As a temporary workaround you can download yabridge from my build from my homepage. I will remove this link as soon as developer makes new release.
https://www.skygge.com/pliki-do-pobrania

**Comment by eugenialoli** [comment]
Thank you! Which file do I need to copy on my ~/.vst3/ folder for yabridge's windows plugins to show up on my DAW? Is it the vst3-chainloader.so file?

**Comment by Skygge666** [comment]
List of files included:
libyabridge-chainloader-clap.so
libyabridge-chainloader-vst2.so
libyabridge-chainloader-vst3.so
libyabridge-clap.so
libyabridge-vst2.so
libyabridge-vst3.so
README.md
yabridgectl
yabridge-host.exe
yabridge-host.exe.so

You can install this the same way as usual binary release. If you have installed yabridge already - just replace these files and run
yabridgectl sync

**Comment by gitterdude** [comment]
Skygge's version isn't compiled against libdl, librt and libpthreads. Maybe that's why it works.

```
ldd .local/share/yabridge-skygge/libyabridge-vst3.so  :
	linux-vdso.so.1 (0x00007874de582000)
	libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007874de000000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007874de2db000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007874de2ad000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007874ddc00000)
	/lib64/ld-linux-x86-64.so.2 (0x00007874de584000)

ldd .local/share/yabridge-org/libyabridge-vst3.so :
	linux-vdso.so.1 (0x000070317c53b000)
	libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x000070317c000000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x000070317c28c000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x000070317bfd2000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x000070317bc00000)
	/lib64/ld-linux-x86-64.so.2 (0x000070317c53d000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x000070317c37e000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x000070317c379000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x000070317c285000)
```

**Comment by erroreyes** [comment]
> List of files included:
> 
> ```
> 
> libyabridge-chainloader-clap.so
> libyabridge-chainloader-vst2.so
> libyabridge-chainloader-vst3.so
> libyabridge-clap.so
> libyabridge-vst2.so
> libyabridge-vst3.so
> README.md
> yabridgectl
> yabridge-host.exe
> yabridge-host.exe.so
> ```
> 
> You can install this the same way as usual binary release. If you have installed yabridge already - just replace these files and run yabridgectl sync

I love you!

PS, Renoise is also another DAW where plugins work without any issues after an update to 24.10, even when adding new VST2 and VST3 plugins.

**Comment by erroreyes** [comment]
I build yabridge on my laptop, including the 32-bit host, updated all my plugins, and everything seems to be working fine. 

[yabridge-error.eyes.zip](https://github.com/user-attachments/files/17706869/yabridge-error.eyes.zip)

**Comment by robbert-vdh** [comment]
> When you try to load the plugin, Yabridge isn't even called. Ardour is probably in a loop trying to load somehow, which increases memory and CPU use.

@paulopinheiro Could you check if that's what's happening? Since the yabridge host application seems to be launching just fine, can you run Ardour from the command line with `YABRIDGE_DEBUG_FILE=/tmp/yabridge.log` set and then check in `/tmp/yabridge.log` to see if the host is starting multiple times? Also, does this only happen with the Ardour binaries from ardour.org or also with Ardour 8.6 installed through Ubuntu's repos?

**Comment by robbert-vdh** [comment]
The problem is likely indeed the line from the backtrace. Weirdly enough there's no problem if you run Ardour's plugin scanner standalone. But when run from Ardour, the `posix_spawn_file_actions_addclose()` calls seem to somehow be both a lot slower then usual, and they somehow leak memory. The reason why a version compiled on Ubuntu 24.10 (or anything else that has glibc 2.34 or newer, like Ubuntu 22.04) is that when yabridge is compiled on one of those distros, it uses a more efficient way to close file descriptors:

https://github.com/robbert-vdh/yabridge/blob/d9286758c05ae4a1d1385c36b5d041dc84b21b76/src/common/process.cpp#L352-L364

I don't know what's changed in Ubuntu 24.10 that's causing `posix_spawn_file_actions_addclose()` to leak memory in the way it does, but the workaround for this would be to move the check if `posix_spawn_file_actions_addclosefrom_np()` is available at runtime and use that if it is. I'll try to find some time to work on that soon.

**Comment by Skygge666** [comment]
Hi, the 5.1.1 seems to be fixed. Thanks!

**Comment by erroreyes** [comment]
Note that I did first encounter this issue with 5.11 in Bitwig.

**Comment by Skygge666** [comment]
5.1.1 works for me also in Bitwig. 5.1.1 was released last week. Maybe you mean 5.1.0?
![obraz](https://github.com/user-attachments/assets/83e93d7a-d8ec-46ae-8e65-e882d69c1d71)

**Comment by erroreyes** [comment]
Not for me. My timeline is that I had 5.1.0 installed on Kubuntu 24.04. I
installed 5.1.1 and it worked. I then upgraded to Kubuntu 24.10 and my
Windows plugins stopped working on Bitwig. I downgraded to 5.1.0 and still
did not work. I built the latest on my laptop under Kubuntu 14.10 and that
worked.

**Comment by gitterdude** [comment]
5.1.1 still doesn't work here in Ardour on Xubuntu 24.10. 
Neither the official release nor the one I compiled myself in Ubuntu 20.04

The self-compiled version from Ubuntu 24.04 works, as 5.1.0 also did.

**Comment by paulopinheiro** [comment]
> I built yabridge on my laptop, including the 32-bit host, updated all my plugins, and everything seems to be working fine.
> 
> [yabridge-error.eyes.zip](https://github.com/user-attachments/files/17706869/yabridge-error.eyes.zip)

This one works for me on UbuntuStudio 24.10
Thank you very much.

**Comment by Skygge666** [comment]
It works for me, thanks a lot! _(Of course under wine-staging 9.21 max, 9.22 is broken.)_

**Comment by gitterdude** [comment]
Works here as well.
I compiled it with gcc-11 on Ubuntu 20.04, transferred the binaries to 24.10 and the VST plugins load in Ardour there.

**Comment by duncte123** [comment]
I can confirm that most of my plugins work fine again with this version

**Comment by Skygge666** [comment]
Closing. Issue resolved.

**Comment by amarandon** [comment]
@robbert-vdh I also bumped into this issue and I confirm that it's fixed by development build 5.1.1-7-gca534308 as you pointed in https://github.com/robbert-vdh/yabridge/issues/377#issuecomment-2504933823

**Comment by DaforLynx** [comment]
Hmm..plugins have the offset issue when using the latest commit. This commit https://github.com/robbert-vdh/yabridge/pull/428/commits/23e3de5737724c53c0262d0e77ef0382d58d90b7 is the last one that worked perfectly for me, I believe.
